src/util-file.c

```diff
@@ -297,6 +297,13 @@ void FileApplyTxFlags(const AppLayerTxData *txd, const uint8_t direction, File *
     uint16_t update_flags = FileFlowFlagsToFlags(txd->file_flags, direction);
     DEBUG_VALIDATE_BUG_ON(
             (file->flags & (FILE_STORE | FILE_NOSTORE)) == (FILE_STORE | FILE_NOSTORE));
+    // May happen with keyword filestore:both,flow :
+    // There may be one opened unclosed file in one direction without filestore
+    // As such it has file->flags & FILE_NOSTORE
+    // But a new file in the other direction may trigger filestore:both,flow
+    // And thus set txd->file_flags & FILE_STORE
```

Comment:
Why don't we store file flags per direction?

Reply:
Looks like this was removed by 79499e4769799e6b4426cace062471a807d64d49, so this should be another bug/ticket...

Reply:
I created this: https://redmine.openinfosecfoundation.org/issues/6617
Please set the version, assignee etc as you see fit. Thank you :)

Reply:
I reworded the title.
Would you want to create a S-V test for it ?
using `filestore:to_server,flow;` and checking that files to_client do not get stored ?

Reply:
Please check https://github.com/OISF/suricata-verify/pull/1556